js/promises/promises.go

```diff
@@ -0,0 +1,46 @@
+// Package promises provides helpers for working with promises in k6.
+package promises
+
+import (
+	"github.com/dop251/goja"
+	"go.k6.io/k6/js/modules"
+)
+
+// MakeHandledPromise can be used to create promises that will be dispatched to k6's event loop.
+//
+// Calling the function will create a goja promise and return its `resolve` and `reject` callbacks, wrapped
+// in such a way that it will block the k6 JS runtime's event loop from exiting before they are
+// called, even if the promise isn't revoled by the time the current script ends executing.
```

Comment:
The current documentation:
1. does not explain why you would use that instead of `goja.NewRuntime()` 
2. Explains things that arguably are better left explain in the eventloop package - and they are.


I would propose that we do actually document *why* you will want to use this and simple say it uses the underlying eventloop mechanism.

Reply:
Good point 👍🏻 Will adapt it indeed 🚀

Reply:
Here's an attempt at improving the documentation accordingly:

```
// MakeHandledPromise facilitates the creation of promises within k6 modules.
//
// Whenever a k6 module function or methods returns a promise, it very likely
// should be created using this function as it ensures the promise is correctly
// managed by k6's event loop.
//
// The function returns a promise, a resolve function and a reject function. The
// promise is the one that should be returned by the function, and the `resolve`
// and `reject` functions should be used to resolve or reject the promise.
//
// Usage example:
//
//	   func myAsynchronousFunction(vu modules.VU) *goja.Promise {
//		    promise, resolve, reject := CreateAsyncFunctionPromise(vu)
//		    go func() {
//		        value, err := someAsynchronousOperation()
//				   if err != nil {
//		            reject(err)
//		            return
//		        }
//
//		        resolve(value)
//		    }()
//		    return promise
//		  }
//
// In the above example, `someAsynchronousOperation` represents an operation that might take a while to finish,
// and k6 script execution shouldn't be halted during its execution. This function ensures such an operation
// is correctly managed within k6's event loop.
```

WDYT? Suggestions to improve?